crm_claim_rma/wizard/claim_make_picking.py

```diff
@@ -0,0 +1,279 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright 2015 Eezee-It
+#    Copyright 2013 Camptocamp
+#    Copyright 2009-2013 Akretion,
+#    Author: Emmanuel Samyn, Raphaël Valyi, Sébastien Beau,
+#            Benoît Guillot, Joel Grand-Guillaume
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.models import api, TransientModel, _
+from openerp.fields import Many2many, Many2one
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from openerp.exceptions import Warning
+from openerp import workflow
+from openerp import workflow
+
+import time
+
+
+class ClaimMakePicking(TransientModel):
+    _name = 'claim_make_picking.wizard'
+    _description = 'Wizard to create pickings from claim lines'
+
+    # Get default source location
+    def _get_source_loc(self):
+        loc_id = False
+        context = self.env.context
+        if context is None:
+            context = {}
+
+        warehouse_obj = self.env['stock.warehouse']
+        warehouse_id = context.get('warehouse_id')
+        picking_type = context.get('picking_type')
+        partner_id = context.get('partner_id')
+        if picking_type == 'out':
+            loc_id = warehouse_obj.browse(
+                warehouse_id).lot_stock_id.id
+        elif partner_id:
+            loc_id = self.env['res.partner'].browse(
+                partner_id).property_stock_customer.id
+
+        return loc_id
+
+    def _get_common_dest_location_from_line(self, line_ids):
+        """Return the ID of the common location between all lines. If no common
+        destination was  found, return False"""
+        loc_id = False
+        line_obj = self.env['claim.line']
+        line_location = []
+
+        for line in line_obj.browse(line_ids):
+            if line.location_dest_id.id not in line_location:
+                line_location.append(line.location_dest_id.id)
+
+        if len(line_location) == 1:
+            loc_id = line_location[0]
+
+        return loc_id
+
+    # Get default destination location
+    def _get_dest_loc(self):
```

Comment:
@aristobulo-eezee I think that this method should have the @api.model because it is called from a field of API 8.0 and the method without @api.model not working. Did you do a functional test in a test instance about this method for default of  claim_line_dest_location field?

After the method have @api.model, the method must return records. So, loc_id = False, must be loc_id = self.env['stock.location'] .